FHEM/00_SIGNALduino.pm

```diff
@@ -2883,84 +2881,99 @@ sub SIGNALduino_IdList($@)
+		}
+		if (length($blacklist) > 0) {							# Blacklist in Hash wandeln
+			SIGNALduino_Log3 $name, 3, "$name IdList: attr blacklistIds=$blacklist";
 			%BlacklistIDs = map { $_ => 1 } split(",", $blacklist);
-			my $w = join ', ' => map "$_" => keys %BlacklistIDs;
-			SIGNALduino_Log3 $name, 4, "$name Attr blacklist $w";
+			#my $w = join ', ' => map "$_" => keys %BlacklistIDs;
+			#SIGNALduino_Log3 $name, 3, "$name IdList, Attr blacklist $w";
 			$bflag = 1;
 		}
 	}
 	
 	my $id;
+	my $wIdFound;						# 1 = Id wurde in der whitelist gefunden
 	foreach $id (keys %ProtocolListSIGNALduino)
 	{
-		next if ($id eq 'id');
-		if ($wflag == 1 && !defined($WhitelistIDs{$id}))
-		{
-			#SIGNALduino_Log3 $name, 3, "skip ID $id";
-			next;
-		}
-		if ($bflag == 1 && defined($BlacklistIDs{$id}))
-		{
-			#SIGNALduino_Log3 $name, 4, "$name skip Blacklist ID $id";
-			next;
-		}
+		$wIdFound = 0;
 		
-		if (defined($ProtocolListSIGNALduino{$id}{developId}) && substr($ProtocolListSIGNALduino{$id}{developId},0,1) eq "m") {
-			my $devid = "m$id";
-			if ($develop !~ m/$devid/) {		# skip wenn die Id nicht im Attribut development steht
-				#SIGNALduino_Log3 $name, 3, "$name: ID=$devid skipped (developId=m)";
-				push (@skippedId, $devid);
+		if ($wflag == 1)				# whitelist
+		{
+			if (defined($WhitelistIDs{$id}))	# Id wurde in der whitelist gefunden
+			{
+				$wIdFound = 1;
+			}
+			else
+			{
+				#Log3 $name, 3, "skip ID $id";
 				next;
 			}
-		}																					 
+		}
 		
-		if (defined($ProtocolListSIGNALduino{$id}{developId}) && substr($ProtocolListSIGNALduino{$id}{developId},0,1) eq "p") {
-			my $devid = "p$id";
-			if ($develop !~ m/$devid/) {		# skip wenn die Id nicht im Attribut development steht
-				#SIGNALduino_Log3 $name, 3, "$name: ID=$devid skipped (developId=p)";
-				push (@skippedId, $devid);
+		if ($bflag == 1 && defined($BlacklistIDs{$id})) {
+				#SIGNALduino_Log3 $name, 3, "$name IdList, skip Blacklist ID $id";
+				push (@skippedBlackId, $id);
 				next;
-			}
 		}
 		
-		if (defined($ProtocolListSIGNALduino{$id}{developId}) && substr($ProtocolListSIGNALduino{$id}{developId},0,1) eq "y") {
-			my $devid = "y$id";
-			if ($develop !~ m/$devid/) {		# skip wenn die Id nicht im Attribut development steht
-				#SIGNALduino_Log3 $name, 3, "$name: ID=$id skipped (developId=y)";
-				push (@skippedId, $devid);
+		# wenn die Id in der whitelist gefunden wurde, es keine developId gibt, dann die folgenden Abfragen ueberspringen
+		if ($wIdFound == 0 && defined($ProtocolListSIGNALduino{$id}{developId}))
+		{
+			if ($ProtocolListSIGNALduino{$id}{developId} eq "m") {
+				if ($develop eq "" || $develop !~ m/m$id/) {  # ist nur zur Abwaertskompatibilitaet und kann in einer der naechsten Versionen entfernt werden
+					push (@devModulId, $id);
+				}
+			}
+			elsif ($ProtocolListSIGNALduino{$id}{developId} eq "p") {
+				SIGNALduino_Log3 $name, 3, "$name IdList: ID=$id skiped (developId=p), not for user";
```

Comment:
Ich bin hier immer noch dafür, die Ausgabe so zu benennen, dass der Anwender erfährt, wofür er es verwenden soll anstelle dass er es überhaupt nicht verwenden soll.

Reply:
Er soll die ID gar nicht verwenden, außer wenn wir ihn dazu bitten um was zu testen.
Wenn er eine ID mit p verwendet ohne, daß wir ihn dazu bitten, kann im Extremfall sein fhem abstürzen.

Reply:
An welcher stelle soll da denn etwas abstürzen? Den von dir vermuteten Fall einer postdemodulation habe ich schon abgefangen.

Reply:
wie dem auch sei, dann wäre besser wenn da steht  cauton, protocol can cause crashes, use only if advised to do.

Reply:
ja, ist so auch ok.

Reply:
Gibt es außer der log Meldung noch mehr was nicht passt?
Dann commite ich es, oder committest Du es.

Reply:
ich schau noch durch woran der Test alles hängt. Kannst es gerne comitten